File: Data Structures & Algorithms/simplify-path/submission-1.java
class Solution {
    public String simplifyPath(String path) {
        Stack<String> pending = new Stack<>();
        boolean isSlash = false;
        String cur = "";
        String res = "";
        for(int i = 0; i < path.length(); ++i) {
            if(path.charAt(i) == '/') {
                if(i > 0 && path.charAt(i - 1) != '/') {
                    if(cur.equals("..")) {
                        if(pending.size() > 0) {
                            pending.pop();
                        }
                    } else if(!cur.equals(".") && cur.length() != 0) {
                        pending.push(cur);
                    }
                }
                
                cur = "";
                continue;
            }

            cur += path.charAt(i);
            if (i == path.length() - 1 && cur.length() != 0) {
                if(cur.equals("..")) {
                    if(pending.size() > 0) {
                        pending.pop();
                    }
                } else if(!cur.equals(".") && cur.length() != 0) {
                    pending.push(cur);
                }
            }
        }
        
        while(!pending.empty()) {
            res = "/" + pending.peek() + res;
            pending.pop();
        }
        if(res.length() == 0) return "/";
        return res;
    }
}

File: Data Structures & Algorithms/count-number-of-islands/submission-1.cpp
class Solution {
public:
    void scan(vector<vector<char>>& grid, int r, int c) {
        if(r < 0 || c < 0 || r >= grid.size() || c >= grid[0].size() || grid[r][c] == '0') return;
        grid[r][c] = '0';
        scan(grid, r + 1, c);
        scan(grid, r - 1, c);
        scan(grid, r, c + 1);
        scan(grid, r, c - 1);

    }
    int numIslands(vector<vector<char>>& grid) {
        stack<pair<int,int>> pending;
        for(int i = 0; i < grid.size(); ++i) {
            for(int j = 0; j < grid[i].size(); ++j) {
                if(grid[i][j] == '1') {
                    pending.emplace(i,j);
                }
            }
        }
        int res = 0;
        while(!pending.empty()) {
            pair<int,int> cur = pending.top();
            pending.pop();
            if(grid[cur.first][cur.second] == '1') {
                ++res;
                scan(grid, cur.first, cur.second);
            }
        }
        return res;
    }
};
